src/object.js

```diff
@@ -228,12 +228,18 @@ export let aliasIn = _.curry((x, prop) => _.getOr(prop, prop, x))
 /**
  * A `_.get` that takes an array of paths (or functions to return values) and returns the value at the first path that matches. Similar to `_.overSome`, but returns the first result that matches instead of just truthy (and supports a default value)
  */
-export let cascade = _.curryN(2, (paths, obj, defaultValue) =>
-  _.flow(
-    findApply((x) => x && _.iteratee(x)(obj)),
-    _.defaultTo(defaultValue)
-  )(paths)
-)
+export let cascade = (paths, obj, defaultValue) =>  _.reduce(
+  (acc, path) => {
+    if (_.isFunction(path)) {
+      const r = path(obj);
+      return r ? _.identity(r) : acc;
+    }
+    const o = _.get(path, obj);
+    return _.isUndefined(o) ? acc : _.identity(o);
+  },
+  defaultValue,
+  paths
+);
```

Comment:
[This fix](https://runkit.com/stellarhoof/futil-issue-339) is much simpler

Reply:
Additionally, this change omits `curryN(2)`, which changes the signature of the method and introduce a bug. We should probably add a test case for that behavior.

Reply:
I wonder if this should be considered a bug in `findApply` and fixed there. That method is just based on `_.find`, which lodash is explicit about matching truthy values and not non-nil values, so my instinct is that `findApply` is probably behaving as it should.

Reply:
> We should probably add a test case for that behavior.

Uff, writing test cases for currying? We'd have to do that for every curried fn in futil.

> However we might want to add an alternatively findApply that works differently on undefined/null and then base this on that method instead.

`findApply` is used once in futil and it's not used at all anywhere else in our repos so I think it'd just add more API surface for a fn that is probably not going to be used